src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,290 @@
+package teammates.storage.api;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.test.BaseTestCaseWithLocalDatabaseAccess;
+
+/**
+ * SUT: {@link NotificationsDb}.
+ */
+public class NotificationsDbTest extends BaseTestCaseWithLocalDatabaseAccess {
+
+    private final NotificationsDb notificationsDb = NotificationsDb.inst();
+    private final Map<String, NotificationAttributes> typicalNotifications = getTypicalDataBundle().notifications;
+
+    @BeforeMethod
+    public void setup() throws Exception {
+        for (NotificationAttributes n : typicalNotifications.values()) {
+            notificationsDb.createEntity(n);
+        }
+    }
+
+    /**
+     * Removes all notifications created by each test.
+     */
+    @AfterMethod
+    public void cleanUp() {
+        List<NotificationAttributes> retrieved = notificationsDb.getAllNotifications();
+        retrieved.forEach(n -> notificationsDb.deleteNotification(n.getNotificationId()));
+    }
+
+    @Test
+    public void testGetNotification() throws Exception {
+        NotificationAttributes n = typicalNotifications.get("notification1");
+
+        ______TS("typical success case");
+        NotificationAttributes actual = notificationsDb.getNotification(n.getNotificationId());
+        assertNotNull(actual);
+
+        ______TS("expect null for non-existent account");
+        actual = notificationsDb.getNotification("invalid_notification_id");
+        assertNull(actual);
+
+        ______TS("failure: null parameter");
+        assertThrows(AssertionError.class, () -> notificationsDb.getNotification(null));
+    }
+
+    @Test
+    public void testGetAllNotifications() throws Exception {
+        ______TS("typical success case");
+        List<NotificationAttributes> actual = notificationsDb.getAllNotifications();
+
+        assertNotNull(actual);
+        typicalNotifications.values().forEach(n -> {
+            if (actual.contains(n)) {
+                actual.remove(n);
+            } else {
+                assertTrue(false);
+            }
```

Comment:
```suggestion
            assertTrue(actual.contains(n));
            actual.remove(n);
```

Reply:
Thank you for the comment. It has been updated.